hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/NodeDecommissionManager.java

```diff
@@ -431,8 +435,9 @@ private synchronized boolean checkIfDecommissionPossible(List<DatanodeDetails> d
           int reqNodes = cif.getReplicationConfig().getRequiredNodes();
           if ((inServiceTotal - numDecom) < reqNodes) {
             String errorMsg = "Insufficient nodes. Tried to decommission " + dns.size() +
-                " nodes of which " + numDecom + " nodes were valid. Cluster has " + inServiceTotal +
-                " IN-SERVICE nodes, " + reqNodes + " of which are required for minimum replication. ";
+                " nodes out of " + inServiceTotal + " IN-SERVICE HEALTHY and " + unHealthyTotal +
+                " UNHEALTHY nodes. Cannot decommission as a minimum of " + reqNodes +
```

Comment:
I think we can also show the number of nodes that are not in service, i.e., DECOMMISSIONING, DECOMMISSIONED, ENTERING_MAINTENANCE, IN_MAINTENANCE to show the full picture to the user. Something like this:
```suggestion
                " IN_SERVICE UNHEALTHY and "  + notInService + " decommissioned/in-maintenance nodes. Cannot decommission as a minimum of " + reqNodes +
```

Reply:
This is all subjective, but I'm cautious about showing a lot of information... it could just end up confusing the user. What do you think about simply showing the number of good nodes (in-service healthy) and the number of bad nodes (not in-service healthy)?

Reply:
That makes sense.. My thought was basically to show 2 numbers: 
1. the number of datanodes that technically can be decommissioned (without the fail-early logic), so the in-service healthy ones. Which is what you have referred to as good nodes above
2. the number of datanodes that cannot be decommissioned (without the fail-early logic), so the unhealthy and not-in-service nodes

If we don't show the unhealthy number, then the issue for which the jira was raised wouldn't be resolved. Maybe as a part of the bad nodes we can show  (not in-service healthy + in-service unhealthy) number of nodes?

Alternatively, we can have a message like this too:
```
"Insufficient nodes. Tried to decommission " + dns.size() + 
" nodes, of which " + numDecom " are IN_SERVICE. " +  
"Cluster has " + inServiceTotal + " IN-SERVICE HEALTHY and " + unHealthyTotal + " UNHEALTHY nodes. " + 
"Cannot decommission as a minimum of " + reqNodes +
" IN-SERVICE HEALTHY nodes are required in the cluster to maintain replication after decommission. ";
```
What do you think about this?

Reply:
Yes, your latest comment is what I meant as well. I agree. One way to get unhealthyTotal is what I commented earlier - https://github.com/apache/ozone/pull/7155#discussion_r1745951180.